Sources/NetworkService/Request.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/// Encapsulates an HTTP request with the necessary components.
+public struct Request {
+    public typealias HTTPHeaderKey = String
```

Comment:
Why the typealias is located within the `Request` type?
Can we have them in a separate file that describes `HTTPHeaders`?

Reply:
Fixed